test/UnitSocketLifecycle.cpp

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright the Collabora Online contributors.
+ *
+ * SPDX-License-Identifier: MPL-2.0
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#include <config.h>
+
+#include <memory>
+#include <string>
+
+#include <HttpRequest.hpp>
+#include <Socket.hpp>
+
+#include <Poco/DOM/DOMParser.h>
+#include <Poco/DOM/Document.h>
+#include <Poco/DOM/Node.h>
+#include <Poco/DOM/NodeFilter.h>
+#include <Poco/DOM/NodeIterator.h>
+#include <Poco/SAX/InputSource.h>
+#include <Poco/URI.h>
+#include <test/lokassert.hpp>
+
+#include <Unit.hpp>
+#include <UserMessages.hpp>
+#include <Util.hpp>
+#include <helpers.hpp>
+#include <vector>
+
+/// Test suite that uses a HTTP session (and not just a socket) directly.
+class UnitSocketLifecycle : public UnitWSD
+{
+    TestResult testOneHTTPSession();
+    TestResult testWebsockNativePingPong();
+    TestResult testWebsockWSDChatPingPong();
+
+public:
+    UnitSocketLifecycle()
+        : UnitWSD("UnitSocketLifecycle")
+    {
+    }
+
+    void invokeWSDTest() override;
+};
+
+UnitBase::TestResult UnitSocketLifecycle::testOneHTTPSession()
+{
+    setTestname(__func__);
+    TST_LOG("Starting Test: " << testname);
+
+    const std::string documentURL = "/favicon.ico";
+
+    // Keep alive socket, avoid forced socket disconnect via dtor
+    TerminatingPoll socketPoller(testname);
+    socketPoller.runOnClientThread();
+
+    const int sockCount = 1;
+    // Reused http session, keep-alive
+    std::vector<std::shared_ptr<http::Session>>
+        sessions; //  = http::Session::create(helpers::getTestServerURI());
+
+    try
+    {
+        for(int sockIdx = 0; sockIdx < sockCount; ++sockIdx) {
+            sessions.push_back( http::Session::create(helpers::getTestServerURI()) );
+            if (sockIdx > 0)
+            {
+                LOK_ASSERT_EQUAL(true, sessions[sockIdx]->isConnected());
+            }
+            TST_LOG("Test: " << testname << "[" << sockIdx << "]: `" << documentURL << "`");
+            http::Request request(documentURL, http::Request::VERB_GET);
+            const std::shared_ptr<const http::Response> response =
+                sessions[sockIdx]->syncRequest(request, socketPoller);
+            TST_LOG("Response: " << response->header().toString());
+            TST_LOG("Response size: " << testname << "[" << sockIdx << "]: `" << documentURL << "`: " << response->header().getContentLength());
+            LOK_ASSERT_EQUAL(http::StatusCode::OK, response->statusCode());
+            LOK_ASSERT_EQUAL(true, sessions[sockIdx]->isConnected());
+            LOK_ASSERT(http::Header::ConnectionToken::None ==
+                       response->header().getConnectionToken());
+            LOK_ASSERT(0 < response->header().getContentLength());
+        }
+    }
+    catch (const Poco::Exception& exc)
```

Comment:
We shouldn't have Poco code here, but perhaps we still do?

Reply:
The unit test is the most unfinished part .. 
So indeed, I might have copy-pasta the POCO exception w/o even using it :)